fundamental-concepts/data-structures-stdlib/2_sums.md

```diff
@@ -0,0 +1,46 @@
+The next fundamental data structure is the enumeration:
+
+<pre>
+data Gender
+  = M
+  | F
+</pre>
+
+Each option is named with an upper-case constructor and preceded by a `|`. An enumeration is just a
+special case of a sum data type:
+
+<pre>
+data Gender
+  = M
+  | F
+  | Other Text
+</pre>
+
+Here, we added a third constructor with a text field. Sum data types can have any number of
+constructors listed with a preceding `|`. Every constructor can take on or more arguments.
+
+When you want to **deconstruct** a sum type in a function, you need to use a **case** statement:
+
+<pre>
+prettyGender: Gender -> Text
+prettyGender gender = case gender of
+  M -> "male"
+  F -> "female"
+  Other o -> o
+</pre>
+
+Or more elegant with a **lambda case** statement:
+
+<pre>
+prettyGender: Gender -> Text
+prettyGender = \case
+  M -> "male"
+  F -> "female"
+  Other o -> o
+</pre>
+
+>> What will "prettyGender M" return ? <<
+=== male
+
+>> What will "prettyGender $ Other "hello world"" return ? <<
```

Comment:
```suggestion
>> What will 'prettyGender $ Other "hello world"' return ? <<
```

Reply:
sadly, backticks don't work in quizzes, let's see how this will look.